src/Scripts/Ecs/Systems/EntityMovementSystem.gd

```diff
@@ -0,0 +1,16 @@
+extends "res://Scripts/Ecs/Core/System.gd"
+
+func on_move_entity(entity, x, y):
+	if is_empty(x, y):
+		entity.x = clamp(x, 0, Constants.TILES_WIDE - 1)
+		entity.y = clamp(y, 0, Constants.TILES_HIGH - 1)
+
+func get_entities_on(x, y):
```

Comment:
Maybe we can rename this to `is_empty` or `is_walkable` or `are_any_entiites_on` - it can return false if found or true at the end. Keeping a list is slightly unnecessary.